### PR TITLE
[MM-14529] Show channel name in post input placeholder

### PR DESCRIPTION
**Pull request by bradjcoughlin**
#### Summary
Displays title of channel in post input placeholder similar to other messaging apps. [Original PR](https://github.com/mattermost/mattermost-webapp/pull/2278) included other features which will be split into another PR.

#### Ticket Link
This was submitted originally for the MatterCon 2019 Hackathon.
https://mattermost.atlassian.net/browse/MM-14529

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates
- [x] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

**Comment by bradjcoughlin** [comment]
@esethna I've submitted this draft PR, the decision about "~" and "@" prefixes notwithstanding.

**Comment by bradjcoughlin** [comment]
Thanks @michaelgamble , I'll open for dev review!

File: components/create_post/create_post.jsx [hunk 1]
```diff
@@ -1113,7 +1113,10 @@ export default class CreatePost extends React.Component {
         if (readOnlyChannel) {
             createMessage = Utils.localizeMessage('create_post.read_only', 'This channel is read-only. Only members with permission can post here.');
         } else {
-            createMessage = Utils.localizeMessage('create_post.write', 'Write a message...');
+            createMessage = formatMessage(
+                {id: 'create_post.write', defaultMessage: 'Write to {channelDisplayName}'},
+                {channelDisplayName: currentChannel.display_name}
```
Comment (saturninoabril) [inline components/create_post/create_post.jsx]: How about for DM/GM channels?  Should it display according to TeammateNameDisplay setting?

Comment (bradjcoughlin) [reply components/create_post/create_post.jsx]: I tested with DMs, but not sure I tested with GMs. DMs should show display name of the team member.

Comment (saturninoabril) [reply components/create_post/create_post.jsx]: Oh right, the `channel.display_name` is a pre-processed field on redux that considers teammate name display setting.

**Pull request merged**